### DISCUSSION TITLE
' Target class [Imdhemy\AppStore\Jws\Parser] does not exist' when receiving apple server notification

### DISCUSSION THREAD

**Discussion by rubanraj7** [imdhemy/laravel-in-app-purchases#414] [Q&A] [answered]
I am getting the following error response, when apple sends the notification to my server. It seems like the package is not available, or am I missing something. I am using version imdhemy/laravel-purchases": "^1.3". 

`Target class [Imdhemy\AppStore\Jws\Parser] does not exist`

**Comment by imdhemy** [comment, marked as answer]
What about one of the following:

1. Upgrading to the [latest release ](https://github.com/imdhemy/laravel-in-app-purchases/releases)of `imdhemy/laravel-purchases`. 
2. Installing the latest version of [imdhemy/appstore-iap](https://github.com/imdhemy/appstore-iap)
3. Upgrading to imdhemy/laravel-purchases `1.4`

Would that help?

**Reply by rubanraj7** [reply]
Let me try to update to 1.4 and get back to you.

Also, I'm assuming that the 'provider' paramenter in the url should be 'provider=app-store-v2' if I'm using using version 2 notifications in apple. 
I wanted to confirm since it's not documented anywhere and found it only because i dug into the code. 
( I can help update the docs after I get this working if you'd like)

**Reply by imdhemy** [reply]
The handler factory recognizes the version based on the request payload. It assumes version 2 if the `signedPayload` exists in the request. 

https://github.com/imdhemy/laravel-in-app-purchases/blob/ff47398cdf29be71bc8f6a632536fcbfa5fbfa99/src/Handlers/HandlerFactory.php#L38-L40

> ( I can help update the docs after I get this working if you'd like)

Thank you! I'd appreciate that.

**Reply by rubanraj7** [reply]
Ok i missed that part..thank you!

I updated to 1.12 and it seems to be working. 

Now I'm just getting a 403 forbidden response, which I'll try to spend some time on.

**Reply by rubanraj7** [reply]
Do i need to store the signature anywhere once i generate the signed url?

**Reply by imdhemy** [reply]
Perfect! 🚀 

[The signature ](https://laravel.com/docs/11.x/urls#signed-urls)depends on the server on which it was created. If you created it on the local server, it won't work on production and vice versa. 

I'd suggest [disabling the signature](https://imdhemy.com/laravel-iap-docs/docs/get-started/routing#validating-a-signed-route-requests) first just to make sure that everything works.

> Do I need to store the signature anywhere once I generate the signed URL?

I don't think so, you can always regenerate a new one and update your Appstore settings.
